source/common/stats/stat_data_allocator_impl.h

```diff
@@ -155,29 +160,30 @@ template <class StatData> class GaugeImpl : public Gauge, public MetricImpl {
   bool used() const override { return data_.flags_ & Flags::Used; }
 
   ImportMode importMode() const override {
-    if (data_.flags_ & Flags::ImportModeUninitialized) {
-      return ImportMode::Uninitialized;
+    if (data_.flags_ & Flags::NeverImport) {
+      return ImportMode::NeverImport;
     } else if (data_.flags_ & Flags::LogicAccumulate) {
       return ImportMode::Accumulate;
     }
-    return ImportMode::NeverImport;
+    return ImportMode::Uninitialized;
   }
 
   void mergeImportMode(ImportMode import_mode) override {
-    if (import_mode != ImportMode::Uninitialized &&
-        (data_.flags_ & Flags::ImportModeUninitialized)) {
-      data_.flags_ &= ~Flags::ImportModeUninitialized;
-      if (import_mode == ImportMode::Accumulate) {
-        data_.flags_ |= Flags::LogicAccumulate;
-      } else {
-        // A previous revision of Envoy must have transferred a gauge that it
-        // thought was Accumulate. But the new version thinks its NeverImport,
-        // so we clear the accumulated value.
-        data_.value_ = 0;
-        data_.flags_ &= ~Flags::Used;
-      }
+    ImportMode current = importMode();
+    if (import_mode == ImportMode::Uninitialized || current == import_mode) {
+      return;
+    }
+
+    ASSERT(current == ImportMode::Uninitialized);
+    if (import_mode == ImportMode::Accumulate) {
+      data_.flags_ |= Flags::LogicAccumulate;
     } else {
-      ASSERT(importMode() == import_mode);
```

Comment:
nit: assert ImportMode::Unused

Reply:
i'm not clear on what you're suggesting here as this is the old code, and there is no ImportMode::Unused; there's an Uninitialized, but I early-exit in that case in the new code.

Reply:
Sorry I was just looking for an assert that would check if there wasn't some other logic flag added later that we are not checking for. I think in this case it must be no import?

`ASSERT(importMode() == ImportMode::NoImport)` ?

Reply:
done; also refactored to switch-statement.